### PR TITLE
Theming

### LINKED ISSUE #274: Merge WIP theme into codebase
This will iron out any issues with merging in UI theme work with the overall codebase. Since we haven't done this part before, it'll take a bit longer than subsequent UI merges.

* [x] Make sure `theming` branch is up to date with `development`
* [ ] Make sure all tests pass in Github workflow

### PR DESCRIPTION
**Pull request by jabrah**
Prefer "Squash and Merge" as Git history in this branch is not important. The UI work is done in a different Github repo, this branch was used by the UI devs to share a Drupal config and snapshots. 

* Adds two Composer repos
  * `jhu-idc/idc-ui-theme` : the theme
  * `jhu-idc/idc-ui-module` : Drupal module that defines some blocks/components to be used by the theme
* Configuration changes (non-exhaustive list)
  * Install custom theme and set as default
  * Custom front page
  * Custom global contact page - includes installing and light configuration of a Captcha module and an associated Image Captcha module
  * Custom taxonomy pages
  * Set global header & footer
  
## Testing

We need to make sure that the theme has been included without disrupting other functionality in IDC.

* Check presence of front page (https://islandora-idc.traefik.me/)
* Check Contact page (https://islandora-idc.traefik.me/contact)
  * Captcha should appear if you are NOT logged in. It should not appear if you are logged in
  * I don't know if email is functional, but this page is configured to send mail to `webmaster@localhost.com` - this is set in Structure > Contact Forms > Website feedback > Edit

### Known issues

Some issues with the included pages:

* Top menu with links to other libraries has a max width. This is not intended, this menu is supposed to extend across the screen
* Inline admin actions tabs (such as View, Edit, etc) have white text color, making it difficult to see
* Contact page should have a picture of a very nice looking building or something. For now we just have a placeholder space
* ~~Item pages don't show info at the moment, will be addressed soon™~~

Closes https://github.com/jhu-idc/iDC-general/issues/274

### CODE REVIEW

File: codebase/composer.json [hunk 1]
```diff
@@ -108,17 +121,24 @@
...
         },
         "installer-paths": {
-            "web/core": ["type:drupal-core"],
-            "web/libraries/{$name}": ["type:drupal-library"],
-            "web/modules/contrib/{$name}": ["type:drupal-module"],
-            "web/profiles/contrib/{$name}": ["type:drupal-profile"],
-            "web/themes/contrib/{$name}": ["type:drupal-theme"],
-            "drush/Commands/contrib/{$name}": ["type:drupal-drush"]
-        },
-        "patches": {
```
Comment (bseeger) [inline codebase/composer.json:118]: As @birkland  pointed out, this should probably be brought back into the code.  (the simplesaml code just below this line)

**Comment by birkland** [comment]
maybe add to known issues list:    I get an error message when viewing admin pages:
```
"layout" not declared in info file. Adaptivetheme requires a compatible layout to be declared in your theme info file e.g. "layout: page-layout". Add the declaration, clear the cache and try again.
What code is in the image? field is required.
```

image

**Comment by birkland** [comment]
Possible issue: Captcha does appear on contact form when I log in as a SAML user.

To replicate:
* on the login page, click the tiny "federated login" link
* when the IDP login page loads, use user `staff1` pass `moo`
* once logged in, navigate to the content page, look for captcha

**Comment by jabrah** [comment]
Noted by @htpvu in Slack, showing an authenticated user a CAPTCHA should be fine for now. The setting can be tuned later if we get pushback during testing.

A site administrator can change who is presented with a CAPTCHA by editing the "Skip CAPTCHA" permission. `Manage > People > Permissions > CAPTCHA > Skip CAPTCHA (Users with this permission will not be offered a CAPTCHA)`

**Comment by jabrah** [comment]
* Revert routing for Islandora Object pages so that they should use default rendering to show data
* The global search widget reverted so search results display using default view
* Update CAPTCHA config to only show CAPTCHA on contact page (previous config would display CAPTCHA in the default search widget for unauthenticated users )

**Review by birkland — APPROVED** [review]
Works as advertised, with known issues noted.  With latest updates, I can at see (un-styled) items and search results now, so it looks hopeful that this shouldn't impede unrelated development effort.  Getting this first iteration in is a nice milestone!

**Pull request merged**